tests/test_primary.py

```diff
@@ -63,6 +71,37 @@ def destroy_temp_dir():
 
 
 
+def create_primary_pinning_file():
+  """ To change the pinned_template.json file to point to the right source for metadata"""
+
+  try:
+    with open(TEST_PINNING_TEMPLATE_FNAME, 'r', encoding = 'utf-8') as pinned_file:
+      pinnings = json.load(pinned_file)
+  except:
+    with codecs.open(TEST_PINNING_TEMPLATE_FNAME, 'r', encoding = 'utf-8') as pinned_file: #Support for Python 2
+      pinnings = json.load(pinned_file)
```

Comment:
There is no need for this span of code, nor the codecs module. Stripping encoding='utf-8' works. If we had to alter the open function, we'd use `from io import open`. Fixing.